src/js/components/TaskEndpointsList.js

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+
+import TaskUtil from '../utils/TaskUtil';
+
+const METHODS_TO_BIND = [
+  'getEndpointsList',
+  'getPortsList',
+  'getShowHideLink',
+  'handlePortsToggle'
+];
+
+class TaskEndpointsList extends React.Component {
```

Comment:
is there any way we can get a `shouldComponentUpdate` on this component? just the fact that it's going to be used on every row of the `TaskTable` makes me feel like this will be extremely helpful if it's possible